### PR TITLE
Adding option to use GT poses as initial values in pose slam

### PR DESCRIPTION
**Pull request by ayushbaid**
The boolean is available in the config files as `use_gt_for_initialization`

### CODE REVIEW

File: gtsfm/pose_slam/pose_slam.py [hunk 1]
```diff
@@ -20,6 +20,27 @@
...
+        initial_values = gtsam.Values()
+        if self._use_gt_for_initialization and gt_wTi_list is not None:
+            logger.info("Using GT for initialization")
+            for i, wTi in enumerate(gt_wTi_list):
+                if wTi is None:
+                    continue
+
+                initial_values.insert(i, wTi)
+
+        initial_values = gtsam.InitializePose3.initialize(pose_init_graph, initial_values, useGradient=False)
```
Comment (dellaert) [inline gtsfm/pose_slam/pose_slam.py:40]: No, since this can fail, I think it has to be either or. So this should be in an else block

**Review by dellaert — APPROVED** [review]
Approve but crucial you fix my comment I think

**Comment by ayushbaid** [comment]
Running into some issue s  where the pose init graph has some keys which are not present in values. Debugging

**Comment by ayushbaid** [comment]
The PR *runs* now, but it has infinity errors:

> Pose SLAM initialization complete with error: inf`
> Pose SLAM optimization complete with error: inf`

**Comment by ayushbaid** [comment]
The final bundle adjustment also has error infinity. Hence this code should *not* be merged.

**Pull request closed without merging**